store/index.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { createContext, useContext } from 'react'
+import { useStaticRendering, observer } from 'mobx-react-lite'
+
+import RootStore from './root'
+
+const isServer = typeof window === 'undefined'
+const GlobalContext = createContext({})
+let globalStore = null
+
+useStaticRendering(isServer)
+
+export const initStore = () => {
+  // on the server-side a new instance is created for each page request
+  // as we don't wan't to mix between users/requests, etc.
+  if (isServer) return new RootStore()
+
+  if (!globalStore) {
+    globalStore = new RootStore()
+    return globalStore
+  }
+
+  return globalStore
+}
+
+export const withGlobalStore = (Component) => {
+  const ObservableComponent = observer(Component)
+  return (props) => {
+    const context = useContext(GlobalContext)
+    return <ObservableComponent store={context} {...props} />
+  }
+}
```

Comment:
This is the initialisation part. We have the `main` component for it. I do not know how to do the context creating and passing at the moment, let's discuss it.

Reply:
I am not sure what do you mean. The initialisation  happens in `main`. These function are part of the store. I would say we use correct place for them.

Reply:
This is not store-related logic. It's pure initialisation. I would move it completely to `main`. If you do not get, let me pick it up later 😉

Reply:
I get what do you mean. I just don't feel there needs to be any action taken. There is nothing wrong with placement at all to my mind.